### ISSUE TITLE
[@typescript-eslint/no-unused-vars-experimental] Got a false positive

### ISSUE THREAD

**Issue by bnoufel** [typescript-eslint/typescript-eslint#4025]
Hi,

This morning i have update my package (see below) after that i go a false positive error on my ts file.

Thank you

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```TS
import babel from "@rollup/plugin-babel";
import commonjs from "@rollup/plugin-commonjs";
import nodeResolve from "@rollup/plugin-node-resolve";
import remove from "rollup-plugin-delete";
import {resolve} from "path";
import {terser} from "rollup-plugin-terser";
import typescript from "@rollup/plugin-typescript";

export default {
    input: "src/EcommerceHttp.ts",
    plugins: [
        remove({targets: resolve("lib", "*")}),
        typescript(),
        commonjs(),
        nodeResolve(),
        babel({
            babelHelpers: "bundled",
            babelrc: false,
            presets: [
                "@babel/preset-env",
                "@babel/preset-typescript"
            ]
        }),
        terser()
    ],
    external: [
        "@benjaminnoufel/http"
    ],
    output: {
        file: resolve("lib", "EcommerceHttp.js"),
        format: "cjs",
        sourcemap: true
    }
};
```

**Expected Result**

No error.

**Actual Result**

```
/home/node/rollup.config.ts
  1:1  error  Definition for rule '@typescript-eslint/no-unused-vars-experimental' was not found  @typescript-eslint/no-unused-vars-experimental
```

**Additional Info**

None.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.0.0` |
| `@typescript-eslint/parser`        | `5.0.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `8.0.1` |
| `node`                             | `16.0.0` |
